R/ard_stack_hierarchical.R

```diff
@@ -0,0 +1,403 @@
+#' Stacked Hierarchical ARD Statistics
+#'
+#' @description
+#' `r lifecycle::badge('experimental')`\cr
+#' Use these functions to calculate multiple summaries of nested or hierarchical data
+#' in a single call.
+#'
+#' - `ard_stack_hierarchical()`: Calculates *rates* of events (e.g. adverse events)
+#'   utilizing the `denominator` and `id` arguments to identify the rows in `data`
+#'   to include in each rate calculation.
+#'
+#' - `ard_stack_hierarchical_count()`: Calculates *counts* of events utilizing
+#'   all rows for each tabulation.
+#'
+#' @details
+#' To calculate event rates, the `ard_stack_hierarchical()` function identifies
+#' rows to include in the calculation.
+#' First, the primary data frame is sorted by the columns identified in
+#' the `id`, `by`, and `variables` arguments.
+#'
+#' As the function cycles over the variables specified in the `variables` argument,
+#' the data frame is grouped by `id`, `intersect(by, names(denominator))`, and `variables`
+#' utilizing the last row within each of the groups.
+#'
+#' @inheritParams ard_hierarchical
+#' @inheritParams ard_stack
+#' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   Specifies the nested/hierarchical structure of the data.
+#'   The variables that are specified here and in the `include` argument
+#'   will have summary statistics calculated.
+#' @param id ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   argument used to subset `data` to identify rows in `data` to calculate
+#'   event rates in `ard_stack_hierarchical()`.
+#' @param denominator (`data.frame`, `integer`)\cr
+#'   used to define the denominator and enhance the output.
+#'   The argument is required for `ard_stack_hierarchical()` and optional
+#'   for `ard_stack_hierarchical_count()`.
+#'   - the univariate tabulations of the `by` variables are calculated with `denominator`,
+#'     when a data frame is passed, e.g. tabulation of the treatment assignment
+#'     counts that may appear in the header of a table.
+#'   - the `denominator` argument must be specified when `id` is used to
+#'     calculate the event rates.
+#'   - if `total_n=TRUE`, the `denominator` argument is used to return the total N
+#' @param include ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   Specify the subset a columns indicated in the `variables` argument for which
+#'   summary statistics will be returned. Default is `everything()`.
+#' @param overall (scalar `logical`)\cr logical indicating whether overall statistics
```

Comment:
The reason I didn't keep them the same is because I was trying to follow the style guidelines, e.g. arguments that appear after the dots`foo(..., .after= TRUE)` begin with a `.`, otherwise they do not. Since these functions don't have the `...`, i dropped it.  BUT this tidyverse guidelines isn't always followed in the tidyverse (but I think that is primarily for historical reasons, rather than intentional style reasons).